src/files/examples/pyfingerprint_examples.py

```diff
@@ -0,0 +1,245 @@
+
+def download_image():
+	"""Downloads image of read fingerprint into temp directory"""
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		print('Downloading image (this take a while)...')
+
+		imageDestination =  tempfile.gettempdir() + '/fingerprint.bmp'
+		f.downloadImage(imageDestination)
+
+		print('The image was saved to "' + imageDestination + '".')
+
+	except Exception as e:
+		logging.exception('Could not download image')
+		exit(1)
+
+def search():
+	"""Searches for given fingerprint"""
+	## Tries to search the finger and calculate hash
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read converting it")
+		f.convertImage(0x01)
+
+		## Searchs template
+		result = f.searchTemplate()
+
+		position_number = result[0]
+		accuracyScore = result[1]
+
+		if ( position_number == -1 ):
+			print('No match found!')
+			exit(0)
+		else:
+			print('Found template at position #' + str(position_number))
+			print('The accuracy score is: ' + str(accuracyScore))
+
+		## OPTIONAL stuff
+		##
+
+		## Loads the found template to charbuffer 1
+		f.loadTemplate(position_number, 0x01)
+
+		## Downloads the characteristics of template loaded in charbuffer 1
+		characteristics = str(f.downloadCharacteristics(0x01)).encode('utf-8')
+		logging.debug("Characteristics raw:%s" % (characteristics))
+
+		## Hashes characteristics of template
+		print('SHA-2 hash of template: ' + hashlib.sha256(characteristics).hexdigest())
+
+	except Exception as e:
+		logging.exception('Could not display index table')
+		exit(1)
+
+def enroll():
+	"""Asks for fingerprint and adds it to database"""
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read, converting it")
+		f.convertImage(0x01)
+
+		## Checks if finger is already enrolled
+		result = f.searchTemplate()
+		position_number = result[0]
+
+		if ( position_number >= 0 ):
+			print('Template already exists at position #' + str(position_number))
+			exit(0)
+
+		print('Remove finger...')
+		time.sleep(2)
+
+		print('Waiting for same finger again...')
+
+		## Wait that finger is read again
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 2
+		f.convertImage(0x02)
+
+		## Compares the charbuffers
+		if ( f.compareCharacteristics() == 0 ):
+			raise Exception('Fingers do not match')
+
+		## Creates a template
+		f.createTemplate()
+
+		## Saves template at new position number
+		position_number = f.storeTemplate()
+		print('Finger enrolled successfully!')
+		print('New template position #' + str(position_number))
+
+	except Exception as e:
```

Comment:
exception message is not printed anymore

Reply:
see next commit, when I call search it seems to be working on my machine, could you investigate why on yours its not the case? do you have a root logger configuration which overrides the default?? 

python /home/pi/dev/rasperry_pi/material/fingerprint_reader/pyfingerprint/src/files/examples/pyfingerprint_examples.py -s
Waiting for finger...
2018-02-16 03:33:23,385 ERROR    Could not display index table
Traceback (most recent call last):
  File "/home/pi/dev/rasperry_pi/material/fingerprint_reader/pyfingerprint/src/files/examples/pyfingerprint_examples.py", line 99, in search
    raise RuntimeError('this is a test exception')
RuntimeError: this is a test exception
Traceback (most recent call last):
  File "/home/pi/dev/rasperry_pi/material/fingerprint_reader/pyfingerprint/src/files/examples/pyfingerprint_examples.py", line 244, in <module>
    main()
  File "/home/pi/dev/rasperry_pi/material/fingerprint_reader/pyfingerprint/src/files/examples/pyfingerprint_examples.py", line 242, in main
    init_arg_parse()
  File "/home/pi/dev/rasperry_pi/material/fingerprint_reader/pyfingerprint/src/files/examples/pyfingerprint_examples.py", line 230, in init_arg_parse
    search()
  File "/home/pi/dev/rasperry_pi/material/fingerprint_reader/pyfingerprint/src/files/examples/pyfingerprint_examples.py", line 99, in search
    raise RuntimeError('this is a test exception')
RuntimeError: this is a test exception

Reply:
Exception message in `e` is not printed. Just append `str(e)` at the end that it looks like `logging.exception('Could not enroll your fingerprint: '+ str(e))`. That gives a better error description what went wrong. Raising the exception will crash the example program. I suggest to using `exit(1)` instead of `raise`.